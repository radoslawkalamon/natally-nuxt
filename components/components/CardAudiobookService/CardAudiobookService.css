.card-audiobook-service {
  align-items: center;
  background-color: var(--color-off-white-1);
  border: var(--border-size) solid var(--color-clickable-border);
  border-radius: var(--border-radius);
  box-shadow: 0 var(--border-size) 0 var(--color-clickable-shadow);
  box-sizing: border-box;
  color: var(--color-off-black-2);
  cursor: pointer;
  display: flex;
  font-family: var(--font-family-sans);
  font-size: var(--gap);
  gap: var(--gap);
  height: var(--height-card-audiobook-service);
  margin: 0;
  max-width: var(--max-width-card-audiobook-service);
  overflow: hidden;
  padding: 1em;
  text-decoration: none;
  width: 100%;
}

.card-audiobook-service__logo-wrapper {
  align-items: center;
  display: flex;
  height: 48px;
  justify-content: center;
  min-width: 48px;
  width: 48px;
}

.card-audiobook-service__logo {
  display: block;
  max-height: 100%;
  max-width: 100%;
}

.card-audiobook-service__cta {
  display: block;
  font-size: var(--font-size-footnotes);
  font-weight: bold;
  line-height: var(--line-height-footnotes);
  margin: 0;
  text-transform: uppercase;
}

.card-audiobook-service__name {
  display: block;
  font-family: var(--font-family-sans);
  font-size: var(--font-size-header-3);
  font-weight: bold;
  line-height: var(--line-height-header);
}

.card-audiobook-service:hover .card-audiobook-service__name {
  text-decoration: underline;
}
