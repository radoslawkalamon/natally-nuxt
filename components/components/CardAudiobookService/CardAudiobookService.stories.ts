import type { Meta, Story } from '@storybook/vue'
import ComponentsCardAudiobookService from '@/components/components/CardAudiobookService/CardAudiobookService.vue'

const meta: Meta = {
  title: 'Components / Card Audiobook Service',
  component: ComponentsCardAudiobookService
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: {
    ComponentsCardAudiobookService
  },
  props: Object.keys(argTypes),
  template: '<ComponentsCardAudiobookService v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  icon: 'https://upload.wikimedia.org/wikipedia/commons/e/e7/Podcasts_%28iOS%29.svg',
  name: 'Apple Podcasts',
  url: 'https://apple.com'
}
