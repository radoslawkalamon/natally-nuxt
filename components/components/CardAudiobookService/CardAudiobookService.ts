import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsCardAudiobookService',
  props: {
    icon: {
      type: String,
      required: true
    },
    name: {
      type: String,
      required: true
    },
    url: {
      type: String,
      required: true
    }
  },
  computed: {
    iconAlt (): string {
      return `Ikona serwisu ${this.name}`
    }
  }
})
