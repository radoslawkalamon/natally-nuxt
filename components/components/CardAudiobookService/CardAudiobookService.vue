<template>
  <a
    class="card-audiobook-service"
    :href="url"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div class="card-audiobook-service__logo-wrapper">
      <img
        :alt="iconAlt"
        class="card-audiobook-service__logo"
        :src="icon"
      >
    </div>
    <div class="card-audiobook-service__meta-wrapper">
      <span class="card-audiobook-service__cta">
        Pos≈Çuchaj w
      </span>
      <span
        class="card-audiobook-service__name"
        v-text="name"
      />
    </div>
  </a>
</template>

<script src="./CardAudiobookService.ts" lang="ts"></script>
<style src="./CardAudiobookService.css" scoped></style>
