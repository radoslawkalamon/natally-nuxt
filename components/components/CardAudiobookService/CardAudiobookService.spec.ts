import { mount } from '@vue/test-utils'
import CardAudiobookService from './CardAudiobookService.vue'
import { shallRender } from '@/devtools/jest.shared.spec'

describe('Components / Card Audiobook Service', () => {
  const defaultOptions = {
    propsData: {
      icon: '/images/podcasts/apple.svg',
      name: 'Apple Podcasts',
      url: 'https://apple.com'
    }
  }

  shallRender(CardAudiobookService, defaultOptions)

  test('shall logo have audiobook service name', () => {
    const wrapper = mount(CardAudiobookService, defaultOptions)
    const logo = wrapper.find('[data-test="components-card-audiobook-service-logo"]')
    const logoAlt = logo.element.getAttribute('alt') || ''
    expect(logoAlt).toContain('Apple Podcasts')
  })
})
