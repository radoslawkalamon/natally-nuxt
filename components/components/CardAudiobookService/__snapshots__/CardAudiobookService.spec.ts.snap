// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Card Audiobook Service Unit shall pass unit sanity test 1`] = `
"<a href="https://apple.com" rel="noopener noreferrer" target="_blank" class="card-audiobook-service">
  <div class="card-audiobook-service__logo-wrapper"><img alt="Ikona serwisu Apple Podcasts" src="/images/podcasts/apple.svg" class="card-audiobook-service__logo"></div>
  <div class="card-audiobook-service__meta-wrapper"><span class="card-audiobook-service__cta">
      Pos≈Çuchaj w
    </span> <span class="card-audiobook-service__name">Apple Podcasts</span></div>
</a>"
`;
